src/lib/pages/backoffice/Acquisition/Order/OrderDetails/OrderLiteratureRequest.js

```diff
@@ -0,0 +1,94 @@
+import { PropTypes } from 'prop-types';
+import React from 'react';
+import { withCancel } from '@api/utils';
+import { documentRequestApi } from '@api/documentRequests/documentRequest';
+import { Link } from 'react-router-dom';
+import { BackOfficeRoutes } from '@routes/urls';
+import { Message } from 'semantic-ui-react';
+import { Loader } from '@components/Loader';
+import _ from 'lodash';
+
+export class OrderLiteratureRequest extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      cancellableLiterature: {},
+      isLoading: true,
+      hasError: false,
+    };
+  }
+
+  componentDidMount() {
+    const {
+      orderId: { pid },
+    } = this.props;
+    this.fetchLiteratureRequest(pid);
+  }
+
+  componentWillUnmount() {
+    this.cancellableLiteratureRequest &&
+      this.cancellableLiteratureRequest.cancel();
+  }
+
+  fetchLiteratureRequest = async (pid) => {
+    try {
+      this.cancellableLiteratureRequest = withCancel(
```

Comment:
do you have a warning here that this.cancellableLiteratureRequest is defined outside the constructor ?

Reply:
No, I don't. I used as a template for the implementation of withCancel other project examples, such as the ones in DocumentPurchaseOrders and DocumentStats.